### ISSUE TITLE
TST: testing not catching bad escape sequences in doc strings

### ISSUE THREAD

**Issue by jklymak** [matplotlib/matplotlib#11335]
### Bug report

Doc strings like

```python
        """
        Initialize the artist inspector with an `Artist` or an iterable of
        `Artist`\s.  If an iterable is used, we assume it is a homogeneous
        sequence (all `Artists` are of the same type) and it is your
        responsibility to make sure this is so.
        """
```

Cause `SyntaxError: invalid escape sequence \s` on my machine.  Somehow they aren't crashing the test suite?

**Comment by timhoffm** [comment]
Please try if this fixes your issue.

**Comment by timhoffm** [comment]
Python 3.6 introduced this as a silent warning https://bugs.python.org/issue27364. Maybe you have different configurations for warnings? That would also be the path to note missing raw strings in our code base.

**Comment by QuLogic** [comment]
`tests.py` turns those [warnings into errors](https://github.com/matplotlib/matplotlib/blob/master/tests.py#L30), but we [stopped using that on Travis](https://github.com/matplotlib/matplotlib/commit/f0d9ca07b85a6868f4777a6861e0608d45ce417c#diff-9230165a077a76ea2cd4ba33bea3c170) a long time ago.

**Comment by tacaswell** [comment]
We should re-enable those warnings on travis (for both 2.2.x and master).

**Comment by jklymak** [comment]
This got fixed....

**Comment by QuLogic** [comment]
Did it? We still don't use `tests.py`, so things are not checked. I fixed several new ones in #11477. We can install the `flake8-invalid-escapes` plugin, or wait until the next minor release of `flake8` which will use `pycodestyle` 2.4.0, that supports this builtin.

**Comment by timhoffm** [comment]
Travis CI `flake8` uses `pycodestyle` 2.5.0. So W605 "invalid escape sequence ‘x'" should be checked.
